Loop/Managers/Alerts/DeviceAlertManager.swift

```diff
@@ -85,19 +89,24 @@ extension DeviceAlertManager: DeviceAlertManagerResponder {
         log.debug("Removing notification %@ from delivered & pending notifications", identifier.value)
         userNotificationCenter.removeDeliveredNotifications(withIdentifiers: [identifier.value])
         userNotificationCenter.removePendingNotificationRequests(withIdentifiers: [identifier.value])
+        alertStore.recordAcknowledgement(of: identifier)
     }
 }
 
 extension DeviceAlertManager: DeviceAlertPresenter {
 
     public func issueAlert(_ alert: DeviceAlert) {
         handlers.forEach { $0.issueAlert(alert) }
+        alertStore.recordIssued(alert: alert)
     }
-    public func removePendingAlert(identifier: DeviceAlert.Identifier) {
-        handlers.forEach { $0.removePendingAlert(identifier: identifier) }
+    
+    public func retractAlert(identifier: DeviceAlert.Identifier) {
+        handlers.forEach { $0.retractAlert(identifier: identifier) }
+        alertStore.recordRetraction(of: identifier)
     }
-    public func removeDeliveredAlert(identifier: DeviceAlert.Identifier) {
-        handlers.forEach { $0.removeDeliveredAlert(identifier: identifier) }
+    
+    private func replayAlert(_ alert: DeviceAlert) {
+        handlers.forEach { $0.issueAlert(alert) }
```

Comment:
[nit] Any need to check to see if the alert is still not in a retracted or acknowledged state?

Reply:
Excellent question!  This (private) function assumes that retraction actually removes the alert from wherever it was issued ... in this case, UserNotificationCenter, which is still where we replay alerts.